scripts/release/goa_prepare/goa_submission.pl

```diff
@@ -274,59 +282,119 @@ sub get_reaction_id {
 
 sub get_proteins_from_catalyst_activity {
     my $parameters = shift;
+
+    my $logger = get_logger(__PACKAGE__);
     
     my $catalyst_activity = $parameters->{'catalyst_activity'};
     my $ontology_letter = $parameters->{'ontology_letter'};
     
-    my @errors = check_catalyst_activity($catalyst_activity);
-    if (@errors) {
-        print(join("\n", @errors) . "\n");
+    my @reasons_to_exclude = check_catalyst_activity($catalyst_activity, $ontology_letter);
+    if (@reasons_to_exclude) {
+        $logger->info(join("\n", @reasons_to_exclude) . "\n");
         return;
-    }          
+    }
     
-    if ($catalyst_activity->activeUnit->[0] && $ontology_letter eq 'F') {
-        return get_proteins_from_physical_entity($catalyst_activity->activeUnit->[0]);
+    if ($ontology_letter eq $MOLECULAR_FUNCTION) {
+        return get_proteins_for_molecular_function(
+            $catalyst_activity->activeUnit->[0] ?
+            $catalyst_activity->activeUnit->[0] :
+            $catalyst_activity->physicalEntity->[0]
+        );
+    } elsif ($ontology_letter eq $BIOLOGICAL_PROCESS) {
+        return get_proteins_for_biological_process($catalyst_activity->physicalEntity->[0]);
     }
-    return get_proteins_from_physical_entity($catalyst_activity->physicalEntity->[0]);
 }
 
 sub check_catalyst_activity {
     my $catalyst_activity = shift;
-    
-    my @errors;
+    my $ontology_letter = shift;
     
     my $catalyst_id = $catalyst_activity->displayName . ' (' . $catalyst_activity->db_id . ')';
     my $physical_entity = $catalyst_activity->physicalEntity->[0];
     my @active_units = @{$catalyst_activity->activeUnit};
+
+    my @reasons_to_exclude;
+    if (!$physical_entity) {
+        push @reasons_to_exclude, "No physical entity: $catalyst_id";
+    } elsif (!$physical_entity->compartment->[0]) {
+        push @reasons_to_exclude, "No compartment for physical entity: $catalyst_id";
+    }
+
+    if ($ontology_letter eq $MOLECULAR_FUNCTION) {
+        push @reasons_to_exclude, "No active unit and physical entity is a complex, set or, polymer: $catalyst_id"
```

Comment:
Everywhere else, it looks like the style is Condition-then-Expression. But here, you have Expression-if-Condition. Could it be changed to be more consistent with the rest of it? It's a bit weird when the layout of conditional expressions and then conditions which they depend upon are suddenly reversed in the middle of a program.

Reply:
Yes, I have actually changed my mind on this after trying it in my younger days :).  I will refactor to use condition followed by expression except in really simple cases.  For things like the following, I'm thinking of keeping it.
`return if $recursion_depth > 2;`